scripts/single-node-all-upgrades.sh

```diff
@@ -147,52 +136,36 @@ upgradeToV3AndV4() {
     echo "Waiting for upgrade to complete..."
     while true; do
         current_version=$(celestia-appd status | jq -r '.node_info.protocol_version.app')
-        if [ "$current_version" = "3" ]; then
-            echo "Upgrade to version 3 complete!"
+        if [ "$current_version" = "${target_version}" ]; then
+            echo "Upgrade to version ${target_version} complete!"
             break
         fi
-        echo "Current version: $current_version, waiting for version 3..."
+        echo "Current version: $current_version, waiting for version ${target_version}..."
         sleep 1
     done
+}
 
-
-    echo "Submitting signal for v4..."
-    celestia-appd tx signal signal 4 \
-        --keyring-backend=${KEYRING_BACKEND} \
-        --home ${APP_HOME} \
-        --from ${KEY_NAME} \
-        --fees ${FEES} \
-        --chain-id ${CHAIN_ID} \
-        --broadcast-mode ${BROADCAST_MODE} \
-        --yes
-
-    sleep 1
-    echo "Querying the tally for v4..."
-    celestia-appd query signal tally 4
-
-    echo "Submitting msg try upgrade..."
-    celestia-appd tx signal try-upgrade \
-        --keyring-backend=${KEYRING_BACKEND} \
-        --home ${APP_HOME} \
-        --from ${KEY_NAME} \
-        --fees ${FEES} \
-        --chain-id ${CHAIN_ID} \
-        --broadcast-mode ${BROADCAST_MODE} \
-        --yes
-
-    echo "Waiting for upgrade to complete..."
+startUpgrades() {
+    sleep 30
+    echo "Waiting for app version 2 before proceeding..."
     while true; do
         current_version=$(celestia-appd status | jq -r '.node_info.protocol_version.app')
-        if [ "$current_version" = "4" ]; then
-            echo "Upgrade to version 4 complete!"
+        if [ "$current_version" = "2" ]; then
+            echo "App version 2 detected, proceeding with v3 upgrade..."
             break
         fi
-        echo "Current version: $current_version, waiting for version 4..."
+        echo "Current version: $current_version, waiting for version 2..."
         sleep 1
     done
+
+    # Perform upgrades to versions 3, 4, 5, and 6
+    performUpgrade 3
+    performUpgrade 4
+    performUpgrade 5
+    performUpgrade 6
```

Comment:
we don't have a v6 yet afiu, so does the upgrade to v6 work? https://github.com/rootulp/celestia-app/blob/562ee1b7d8b7cd6a3aa6849306f7ad68a79758bd/internal/embedding/data.go#L10-L16

Reply:
main is v6: https://github.com/celestiaorg/celestia-app/blob/d8bd2676fdb06dbef783e26ec9c76e1cdcf606eb/go.mod#L1

Reply:
how do we upgrade though if we don't have a version in the multiplexer with v6

Reply:
https://github.com/celestiaorg/celestia-app/blob/d8bd2676fdb06dbef783e26ec9c76e1cdcf606eb/cmd/celestia-appd/cmd/modify_root_command_multiplexer.go#L67-L86

perhaps I'm just misunderstanding how the upgrades work?